### ISSUE TITLE
Issues with Next 13

### ISSUE THREAD

**Issue by holdenmad** [segmentio/consent-manager#303]
I'm trying to use the consent-manager for a Next.js 13 project, but running into problems. I've tried to using [this Next.js example in the documentation](https://www.npmjs.com/package/@segment/consent-manager#example-in-nextjs) in my code, but nothing seems to work and I keep getting this error: 

<img width="1394" alt="Server Error TypeError: Object prototype may only be an Object or null: undefined This error happened while generating the page. Any console logs will be displayed in the terminal window." src="https://user-images.githubusercontent.com/54632553/214796706-26e45bf2-89cf-4d84-8c13-91f532702bb4.png">

The standalone script works if I put it in the Head and use `Script` elements like this: 

```
import Script from "next/script";
import * as snippet from "@segment/snippet";

export default function Head() {
  const loadSegment = () => {
    const options = {
      apiKey: process.env.NEXT_PUBLIC_SEGMENT_WRITE_KEY,
      page: true,
      load: true,
    };
    return snippet.min(options);
  };
  return (
    <>
      <title></title>
      <meta content="width=device-width, initial-scale=1" name="viewport" />
      <link rel="icon" href="/favicon.ico" />
      <Script
        dangerouslySetInnerHTML={{ __html: loadSegment() }}
        id="segmentScript"
      />
      <Script
        id="consentManagerScript"
        dangerouslySetInnerHTML={{
          __html: `window.consentManagerConfig = function(exports) {
            var React = exports.React
            var inEU = exports.inEU

            var bannerContent = React.createElement(
              'span',
              null,
              'We use cookies (and other similar technologies) to collect data to improve your experience on our site. By using our website, you՚re agreeing to the collection of data as described in our',
              ' ',
              React.createElement(
                'a',
                { href: '/docs/legal/website-data-collection-policy/', target: '_blank' },
                'Website Data Collection Policy'
              ),
              '.'
            )
            var bannerSubContent = 'You can change your preferences at any time.'
            var preferencesDialogTitle = 'Website Data Collection Preferences'
            var preferencesDialogContent =
              'We use data collected by cookies and JavaScript libraries to improve your browsing experience, analyze site traffic, deliver personalized advertisements, and increase the overall performance of our site.'
            var cancelDialogTitle = 'Are you sure you want to cancel?'
            var cancelDialogContent =
              'Your preferences have not been saved. By continuing to use our website, you՚re agreeing to our Website Data Collection Policy.'

            return {
              container: '#target-container',
              writeKey: '${process.env.NEXT_PUBLIC_SEGMENT_WRITE_KEY}',
              shouldRequireConsent: inEU,
              bannerContent: bannerContent,
              bannerSubContent: bannerSubContent,
              preferencesDialogTitle: preferencesDialogTitle,
              preferencesDialogContent: preferencesDialogContent,
              cancelDialogTitle: cancelDialogTitle,
              cancelDialogContent: cancelDialogContent
            }
          }`,
        }}
      />
      <Script src="https://unpkg.com/@segment/consent-manager@5.3.0/standalone/consent-manager.js" />
    </>
  );
}
``` 

I'm able to proceed with my project using the standalone script, but I feel there might be an issue with using the `ConsentManager` component in Next 13. Has anyone else had any issues with this? Or does anyone have any solutions that I haven't seen? I'd prefer to use the `ConsentManager` component if I can. 

Version = "next": "13.1.1",

**Comment by edsonjab** [comment]
Hi @holdenmad  thank you for your report, we start looking into this.

**Comment by Vanss472** [comment]
I actually have the same issue using the { ConsentManager, openConsentManager } from "@segment/consent-manager" in NextJs 13.

When I clicked the button the `PreferenceDialog` doesn't open but it does appear in the ReactComponentTree(React Developer Tools)
`<button type="button" onClick={openConsentManager}>Website Data Collection Preferences</button>`

Using the standalone script works fine.

"next": "13.0.2",
"react": "18.2.0"

**Comment by farlandlee** [comment]
@edsonjab any update on this?  I am also trying to use consent manager in Next 13 and the PreferenceDialog does not open.  Ideally we would like to use the Consent Manager Builder to match the UI design on our main site. Thank you!

"@segment/consent-manager": "5.7.0"
"@segment/snippet": "4.15.3"
"next": "13.2.3"
"react": "18.2.0"

**Comment by seanlail** [comment]
I have the same issue with Next 12:

```
    "@segment/consent-manager": "5.7.0",
    "@segment/in-eu": "0.4.0",
    "next": "12.3.1",
    "react": "18.2.0",
```

**Comment by edsonjab** [comment]
Hi @holdenmad, @farlandlee, @seanlail 
Sorry to answer late, I created this  [example ](https://github.com/edsonjab/next_consent-manager) with nextJS, I checked that preference Dialog it doesn't open on mode dev, but if you run build command your project and after run the project with start command, it works, I will check why it doesn't work on dev mode.

**Comment by seanlail** [comment]
> Hi @holdenmad, @farlandlee, @seanlail Sorry to answer late, I created this [example ](https://github.com/edsonjab/next_consent-manager) with nextJS, I checked that preference Dialog it doesn't open on mode dev, but if you run build command your project and after run the project with start command, it works, I will check why it doesn't work on dev mode.

Sounds like double rendering: https://beta.reactjs.org/reference/react/StrictMode#fixing-bugs-found-by-double-rendering-in-development

**Comment by mausic** [comment]
> Hi @holdenmad, @farlandlee, @seanlail Sorry to answer late, I created this [example ](https://github.com/edsonjab/next_consent-manager) with nextJS, I checked that preference Dialog it doesn't open on mode dev, but if you run build command your project and after run the project with start command, it works, I will check why it doesn't work on dev mode.

Hi @seanlail,

Any updates so far?

**Comment by anvar-drewl** [comment]
Hi @edsonjab 
There is still no confirmation block even in built example
<img width="1278" alt="Снимок экрана 2023-03-28 в 19 45 24" src="https://user-images.githubusercontent.com/113582591/228258117-7e545c9b-986b-4ffa-8586-b149a5735c39.png">

any updates on that issue?

**Comment by edsonjab** [comment]
Hi @mausic @anvar-drewl, I found a solution, on `next.config.js` file you need change this `reactStrictMode` property on `false`

![image](https://user-images.githubusercontent.com/22898327/228413367-46654f41-4dd4-4631-8c86-37ec4eb3dda3.png)

**Comment by edsonjab** [comment]
@anvar-drewl we still working on a solution

**Comment by anvar-drewl** [comment]
> Hi @mausic @anvar-drewl, I found a solution, on `next.config.js` file you need change this `reactStrictMode` property on `false`
> 
> ![image](https://user-images.githubusercontent.com/22898327/228413367-46654f41-4dd4-4631-8c86-37ec4eb3dda3.png)

@edsonjab Could you please push your changes to your existing repo? [next_consent-manager](https://github.com/edsonjab/next_consent-manager)

**Comment by seg-leonelsanches** [comment]
> Hi @mausic @anvar-drewl, I found a solution, on `next.config.js` file you need change this `reactStrictMode` property on `false`
> 
> ![image](https://user-images.githubusercontent.com/22898327/228413367-46654f41-4dd4-4631-8c86-37ec4eb3dda3.png)

That worked for me. Thanks!

**Comment by filingroove** [comment]
eh, but what if I want `reactStrictMode: true`? 🤔

**Comment by kairilliet** [comment]
So I'm having a weird issue with my nextjs implementation where if I call openConsentManager() from the browser console, I get the pop up to appear, but the second Cancel Dialog pop up does not render if I click the X or cancel button. It just disappears. I've tried to implement all the changes related to nextjs that I've seen on here but no luck.

A weird side thought: If I use the example segment API write key instead of my segment write key, I do get the CancelDialog popup window showing.

Any thoughts? Using latest version of nextjs and we forked off segment/consent-manager for our own consent manager due to a few things, but it's got all the same bones and structure. Using the standalone consent-manager as well.
